src/game/logic/object/update/aiupdate.h

```diff
@@ -200,21 +242,47 @@ class AIUpdateInterface : public UpdateModule, public AICommandInterface
 
     float Get_Cur_Locomotor_Speed() const;
     bool Get_Turret_Rot_And_Pitch(WhichTurretType tur, float *turret_angle, float *turret_pitch);
+    unsigned int Get_Mood_Matrix_Action_Adjustment(MoodMatrixAction action) const;
+    Object *Get_Next_Mood_Target(bool called_by_ai, bool called_during_idle);
+    void Reset_Next_Mood_Check_Time();
+
     bool Has_Locomotor_For_Surface(LocomotorSurfaceType t);
 
     const Locomotor *Get_Cur_Locomotor() const { return m_curLocomotor; }
     Locomotor *Get_Cur_Locomotor() { return m_curLocomotor; }
     LocomotorSetType Get_Cur_Locomotor_Set() { return m_curLocomotorSet; }
     const Path *Get_Path() const { return m_path; }
     ObjectID Get_Ignored_Obstacle_ID() { return m_ignoreObstacleID; }
+    const AIStateMachine *Get_State_Machine() const { return m_stateMachine; }
+    AIStateMachine *Get_State_Machine() { return m_stateMachine; }
+    unsigned int Get_Next_Mood_Check_Time() const { return m_nextMoodCheckTime; }
+    WhichTurretType Get_Current_Turret() const { return m_currentTurret; }
+    Object *Get_Goal_Object() { return Get_State_Machine()->Get_Goal_Object(); }
+
+    const AIUpdateModuleData *Get_AI_Update_Module_Data() const
+    {
+        return static_cast<const AIUpdateModuleData *>(Get_Module_Data());
+    }
+
+    void Set_Current_Turrt(WhichTurretType t) { m_currentTurret = t; }
+
+    bool Are_Turrets_Linked() const { return Get_AI_Update_Module_Data()->m_turretsLinked; }
 
 private:
-    unsigned char unk[0x10C];
+    unsigned char unk[0x8];
+    AIStateMachine *m_stateMachine;
+    unsigned char unk2[0x100];
     Path *m_path;
-    unsigned char unk2[0x20];
+    unsigned char unk3[0x20];
     ObjectID m_ignoreObstacleID;
-    unsigned char unk3[0x48];
+    unsigned char unk4[0x48];
     LocomotorSet m_locomotorSet;
     Locomotor *m_curLocomotor;
     LocomotorSetType m_curLocomotorSet;
+    LocoGoalType m_locomotorGoalType;
+    Coord3D m_locomotorGoalData;
+    TurretAI *m_turretAI[2];

```

Comment:
WhichTurretType should have MAX_TURRETS which then us this 2

Reply:
fixed.

Reply:
Also further above at `TurretAIData *m_turretData[2];`

Reply:
fixed.